rotkehlchen/tests/unit/test_data_updates.py

```diff
@@ -467,17 +467,18 @@ def test_update_rpc_nodes(data_updater: RotkiDataUpdater) -> None:
 
     # check the db state of the user's rpc_nodes
     custom_node_tuple = ('custom node', 'https://node.rotki.com/', 1, 1, '0.50', 'ETH')
-    with data_updater.user_db.user_write() as write_cursor:
-        write_cursor.execute('SELECT COUNT(*) FROM rpc_nodes')
-        assert write_cursor.fetchone()[0] == default_rpc_nodes_count
+    with data_updater.user_db.conn.read_ctx() as cursor:
+        cursor.execute('SELECT COUNT(*) FROM rpc_nodes')
+        assert cursor.fetchone()[0] == default_rpc_nodes_count
         # add a custom node.
-        write_cursor.execute(
-            'INSERT INTO rpc_nodes(name, endpoint, owned, active, weight, blockchain) '
-            'VALUES(?, ?, ?, ?, ?, ?)',
-            custom_node_tuple,
-        )
-        write_cursor.execute('SELECT COUNT(*) FROM rpc_nodes')
-        assert write_cursor.fetchone()[0] == default_rpc_nodes_count + 1
+        with data_updater.user_db.conn.write_ctx() as write_cursor:
+            write_cursor.execute(
+                'INSERT INTO rpc_nodes(name, endpoint, owned, active, weight, blockchain) '
+                'VALUES(?, ?, ?, ?, ?, ?)',
+                custom_node_tuple,
+            )
+        cursor.execute('SELECT COUNT(*) FROM rpc_nodes')
```

Comment:
you need to close the cursor before reading to ensure that the other connection can see it right?

Reply:
![Yes.](https://github.com/user-attachments/assets/3318aabe-7067-426b-8dee-ae304a8cedef)